cli: TestZipRetries failed

cli.TestZipRetries [failed](https://tanzanite.cluster.engflow.com/invocations/default/b6e469ac-146e-41cd-aeaf-59619a9f5902?testReportRun=6&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvY2xpOmNsaV90ZXN0) on master @ [c7f16c96d4db6ef97085f972f094c7e4e7c1518f](https://github.com/cockroachdb/cockroach/commits/c7f16c96d4db6ef97085f972f094c7e4e7c1518f):


```
=== RUN   TestZipRetries
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestZipRetries2072478922
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
[test] retrieving SQL data for generate_series(1,15000) as t(x)...
[test] retrieving SQL data for generate_series(1,15000) as t(x): writing output: test/generate_series(1,15000) as t(x).json...
(error encountered after some results were delivered)
[test] retrieving SQL data for generate_series(1,15000) as t(x): last request failed: ERROR: query execution canceled due to statement timeout (SQLSTATE 57014)
[test] retrieving SQL data for generate_series(1,15000) as t(x): creating error output: test/generate_series(1,15000) as t(x).json.err.txt...
[test] retrieving SQL data for generate_series(1,15000) as t(x): done
    zip_test.go:829: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/cli/zip_test.go:829
        	Error:      	Not equal: 
        	            	expected: "test/generate_series(1,15000) as t(x).json\ntest/generate_series(1,15000) as t(x).json.err.txt\ntest/generate_series(1,15000) as t(x).1.json\ntest/generate_series(1,15000) as t(x).1.json.err.txt\ntest/generate_series(1,15000) as t(x).2.json\ntest/generate_series(1,15000) as t(x).2.json.err.txt\ntest/generate_series(1,15000) as t(x).3.json\ntest/generate_series(1,15000) as t(x).3.json.err.txt\ntest/generate_series(1,15000) as t(x).4.json\ntest/generate_series(1,15000) as t(x).4.json.err.txt\n"
        	            	actual  : "test/generate_series(1,15000) as t(x).json\ntest/generate_series(1,15000) as t(x).json.err.txt\n"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -2,10 +2,2 @@
        	            	 test/generate_series(1,15000) as t(x).json.err.txt
        	            	-test/generate_series(1,15000) as t(x).1.json
        	            	-test/generate_series(1,15000) as t(x).1.json.err.txt
        	            	-test/generate_series(1,15000) as t(x).2.json
        	            	-test/generate_series(1,15000) as t(x).2.json.err.txt
        	            	-test/generate_series(1,15000) as t(x).3.json
        	            	-test/generate_series(1,15000) as t(x).3.json.err.txt
        	            	-test/generate_series(1,15000) as t(x).4.json
        	            	-test/generate_series(1,15000) as t(x).4.json.err.txt
        	            	 
        	Test:       	TestZipRetries
    zip_test.go:830: -- test log scope end --
test logs left over in: outputs.zip/logTestZipRetries2072478922
--- FAIL: TestZipRetries (4.20s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=6</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestZipRetries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-38986

cli.TestZipRetries [failed](https://tanzanite.cluster.engflow.com/invocations/default/b29df2c6-6c4c-4017-aaba-9be355e35afa?testReportRun=2&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvY2xpOmNsaV90ZXN0) on master @ [c7f16c96d4db6ef97085f972f094c7e4e7c1518f](https://github.com/cockroachdb/cockroach/commits/c7f16c96d4db6ef97085f972f094c7e4e7c1518f):


```
=== RUN   TestZipRetries
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestZipRetries521271180
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
[test] retrieving SQL data for generate_series(1,15000) as t(x)...
[test] retrieving SQL data for generate_series(1,15000) as t(x): writing output: test/generate_series(1,15000) as t(x).json...
(error encountered after some results were delivered)
[test] retrieving SQL data for generate_series(1,15000) as t(x): last request failed: ERROR: query execution canceled due to statement timeout (SQLSTATE 57014)
[test] retrieving SQL data for generate_series(1,15000) as t(x): creating error output: test/generate_series(1,15000) as t(x).json.err.txt...
[test] retrieving SQL data for generate_series(1,15000) as t(x): done
    zip_test.go:829: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/cli/zip_test.go:829
        	Error:      	Not equal: 
        	            	expected: "test/generate_series(1,15000) as t(x).json\ntest/generate_series(1,15000) as t(x).json.err.txt\ntest/generate_series(1,15000) as t(x).1.json\ntest/generate_series(1,15000) as t(x).1.json.err.txt\ntest/generate_series(1,15000) as t(x).2.json\ntest/generate_series(1,15000) as t(x).2.json.err.txt\ntest/generate_series(1,15000) as t(x).3.json\ntest/generate_series(1,15000) as t(x).3.json.err.txt\ntest/generate_series(1,15000) as t(x).4.json\ntest/generate_series(1,15000) as t(x).4.json.err.txt\n"
        	            	actual  : "test/generate_series(1,15000) as t(x).json\ntest/generate_series(1,15000) as t(x).json.err.txt\n"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -2,10 +2,2 @@
        	            	 test/generate_series(1,15000) as t(x).json.err.txt
        	            	-test/generate_series(1,15000) as t(x).1.json
        	            	-test/generate_series(1,15000) as t(x).1.json.err.txt
        	            	-test/generate_series(1,15000) as t(x).2.json
        	            	-test/generate_series(1,15000) as t(x).2.json.err.txt
        	            	-test/generate_series(1,15000) as t(x).3.json
        	            	-test/generate_series(1,15000) as t(x).3.json.err.txt
        	            	-test/generate_series(1,15000) as t(x).4.json
        	            	-test/generate_series(1,15000) as t(x).4.json.err.txt
        	            	 
        	Test:       	TestZipRetries
    zip_test.go:830: -- test log scope end --
test logs left over in: outputs.zip/logTestZipRetries521271180
--- FAIL: TestZipRetries (8.43s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestZipRetries.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/cli/zip_test.go
+++ b/pkg/cli/zip_test.go
@@ -750,15 +750,8 @@ func TestZipRetries(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 124650)
-
 	s := serverutils.StartServerOnly(t, base.TestServerArgs{Insecure: true})
 	defer s.Stopper().Stop(context.Background())
-
-	// Lower the buffer size so that an error is returned when running the
-	// generate_series query.
-	_, err := s.SQLConn(t).Exec(`SET CLUSTER SETTING sql.defaults.results_buffer.size = '16kiB'`)
-	require.NoError(t, err)
 
 	dir, cleanupFn := testutils.TempDir(t)
 	defer cleanupFn()
@@ -777,11 +770,13 @@ func TestZipRetries(t *testing.T) {
 			}
 		}()
 
+		// Lower the buffer size so that an error is returned when running the
+		// generate_series query.
 		sqlURL := url.URL{
 			Scheme:   "postgres",
 			User:     url.User(username.RootUser),
 			Host:     s.AdvSQLAddr(),
-			RawQuery: "sslmode=disable",
+			RawQuery: "sslmode=disable&results_buffer_size=16KiB",
 		}
 		sqlConn := sqlConnCtx.MakeSQLConn(io.Discard, io.Discard, sqlURL.String())
 		defer func() {

